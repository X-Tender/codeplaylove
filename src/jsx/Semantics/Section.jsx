import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Article = ({ className, children, isFullWidth }) => {
	const classes = classNames('section', className, { 'is-fullwidth': isFullWidth });
	return <section className={classes}>{children}</section>;
};

Article.propTypes = {
	className: PropTypes.string,
	children: PropTypes.node,
	isFullWidth: PropTypes.bool,
};

Article.defaultProps = {
	className: '',
	children: null,
	isFullWidth: false,
};

export default Article;
