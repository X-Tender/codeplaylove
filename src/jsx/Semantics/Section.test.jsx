import React from 'react';
import { shallow } from 'enzyme';
import Section from './Section';

describe('Section', () => {
	const className = 'section';
	const children = <p>Is this the real life?</p>;

	describe('with all props', () => {
		const comp = shallow(
			<Section className={className} isFullWidth>
				{children}
			</Section>
		);

		it('renders correctly', () => {
			expect(comp).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			expect(comp.find('section').hasClass(className)).toBe(true);
			expect(comp.find('section').hasClass('is-fullwidth')).toBe(true);
			expect(comp.find('section').contains(children)).toBe(true);
		});
	});

	describe('with no props', () => {
		const comp = shallow(<Section />);

		it('renders correctly', () => {
			expect(comp).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			expect(comp.find('section').children()).toHaveLength(0);
		});
	});
});
