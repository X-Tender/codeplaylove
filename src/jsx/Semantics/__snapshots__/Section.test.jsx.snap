// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section with all props renders correctly 1`] = `
<section
  className="section section is-fullwidth"
>
  <p>
    Is this the real life?
  </p>
</section>
`;

exports[`Section with no props renders correctly 1`] = `
<section
  className="section"
/>
`;
